notebooks/generative-ai/chatbot.ipynb

```diff
@@ -0,0 +1,433 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Chatbot with LangChain conversational chain and OpenAI 🤖💬\n",
+    "\n",
+    "<a target=\"_blank\" href=\"https://colab.research.google.com/github/elastic/elasticsearch-labs/blob/main/notebooks/generative-ai/chatbot.ipynb\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>\n",
+    "\n",
+    "In this notebook we'll build a chatbot that can respond to questions about custom data, such as policies of an employer.\n",
+    "\n",
+    "The chatbot uses LangChain's `ConversationalRetrievalChain` and has the following capabilities:\n",
+    "- Answer questions asked in natural language\n",
+    "- Run hybrid search in Elasticsearch to find documents that answer the question\n",
+    "- Extract and summarize the answer using OpenAI LLM\n",
+    "- Maintain conversational memory for follow-up questions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Requirements 🧰\n",
+    "For this example, you will need:\n",
+    "\n",
+    "- Python 3.6 or later\n",
+    "- An Elastic deployment with minimum **2GB machine learning node**\n",
+    "  - We'll be using [Elastic Cloud](https://www.elastic.co/guide/en/cloud/current/ec-getting-started.html) for this example (available with a [free trial](https://cloud.elastic.co/registration))\n",
+    "- OpenAI account\n",
+    "\n",
+    "### Create Elastic Cloud deployment\n",
+    "\n",
+    "If you don't have an Elastic Cloud deployment, follow these steps to create one.\n",
+    "1. Go to https://cloud.elastic.co/registration and sign up for a free trial\n",
+    "2. Select **Create Deployment**\n",
+    "3. Add a 2GB machine learning node to the deployment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install packages 📦\n",
+    "\n",
+    "First we `pip install` the packages we need for this example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%pip install -U langchain==0.0.245 jq openai elasticsearch tiktoken"
```

Comment:
It looks like `jq` and `toktoken` aren't used. I think they can be removed?

I'd also be inclined to either not pin `langchain`'s version, or provide something like `>=` if there is a minimum version requirement.

Reply:
> I'd also be inclined to either not pin langchain's version, or provide something like >= if there is a minimum version requirement.

Unfortunately theres a bug with elasticsearch langchain vectorstore that was introduced recently. My changes (soon to be merged!) will update this code and no longer need to peg the version.

Reply:
Removed `jq` and `tiktoken`.